### PR TITLE
Nada/p2 ps 827/my ads page refactor

### PR DESCRIPTION
**Pull request by nada-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/p2p/src/components/modal-manager/modals/ad-created-modal/ad-created-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                { should_stack_modal: isMobile() }
+            );
+        } else if (advert_details?.visibility_status?.includes(AD_EXCEEDS_DAILY_LIMIT)) {
+            showModal(
+                {
+                    key: 'AdVisibilityErrorModal',
+                    props: { error_code: AD_EXCEEDS_DAILY_LIMIT },
+                },
+                { should_stack_modal: isMobile() }
+            );
```
Comment (adrienne-deriv) [inline packages/p2p/src/components/modal-manager/modals/ad-created-modal/ad-created-modal.tsx:44]: Maybe something like:
```js
if ([AD_EXCEEDS_BALANCE, AD_EXCEEDS_DAILY_LIMIT].includes(advert_details?.visibility_status) {
            showModal(
                {
                    key: 'AdVisibilityErrorModal',
                    props: { error_code: advert_details.visibility_status }
                },
                { should_stack_modal: isMobile() }
            );
}

```

File: packages/p2p/src/pages/my-ads/ad-form-submit/ad-form-submit.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    onCleanup = () => {
+        // do nothing
+    },
+    payment_method_details,
+    payment_method_names,
+    selected_methods,
+}: TAdFormSubmitProps) => {
+    const { dirty, isSubmitting, isValid } = useFormikContext();
+    const check_button = isSubmitting || !isValid || selected_methods.length === 0 || current_method.is_deleted;
```
Comment (adrienne-deriv) [inline packages/p2p/src/pages/my-ads/ad-form-submit/ad-form-submit.tsx:50]: nit: Maybe an alternative naming would be `should_disable_button`?
```suggestion
    const should_disable_button = isSubmitting || !isValid || selected_methods.length === 0 || current_method.is_deleted;
```

File: packages/p2p/src/components/modal-manager/modals/ad-created-modal/ad-created-modal.tsx (hunk 1 above)
Comment (nada-deriv) [reply packages/p2p/src/components/modal-manager/modals/ad-created-modal/ad-created-modal.tsx:44]: advert_details.visibility_status is an array. so i would not be able to pass the specific status. btw the error checking will be removed from this section once advisibilityerrormodal refactoring is done. i have added a todo comment above.

**Review by niloofar-deriv — APPROVED** [review]
LGTM :+1:

**Pull request closed without merging**